[Keyboard] Add Flatbread60, Vaneela, and VaneelaEx by nckiibs

noclew opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

ridingqwerty commented on keyboards/nckiibs/flatbread60/keymaps/default/readme.md:
```diff
@@ -0,0 +1 @@
+# The default keymap for waffle60
```
```suggestion
# The default keymap for flatbread60
```

noclew replied:
Done!

ridingqwerty commented on keyboards/nckiibs/flatbread60/rules.mk:
```diff
@@ -0,0 +1,23 @@
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+# FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+# HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
```suggestion
```

noclew replied:
done!

ridingqwerty commented on keyboards/nckiibs/flatbread60/readme.md:
```diff
@@ -0,0 +1,19 @@
+
+This is a custom PCB made by a hobbiest, and it has an ortholinear 60 key layout. Flatbread60 supports Pro Micro and its wireless variants, such as Blue Micro. A battery switch slot and extra pinout is on the board.  
+
+* Keyboard Maintainer: [noclew](https://github.com/yourusername)
```
```suggestion
* Keyboard Maintainer: [noclew](https://github.com/noclew)
```

noclew replied:
done!

ridingqwerty commented on keyboards/nckiibs/flatbread60/readme.md:
```diff
@@ -0,0 +1,19 @@
+# flatbread60
+
+![flatbread60](imgur.com image replace me!)
```
Let's get an image link here if possible.

noclew replied:
done!

ridingqwerty commented on keyboards/nckiibs/flatbread60/info.json:
```diff
@@ -0,0 +1,253 @@
+{
+    "keyboard_name": "creOnic",
```
```suggestion
    "keyboard_name": "flatbread60",
```

noclew replied:
done

ridingqwerty commented on keyboards/nckiibs/flatbread60/config.h:
```diff
@@ -0,0 +1,251 @@
+#define DEVICE_VER 0x0001
+#define MANUFACTURER noclew
+#define PRODUCT Flatbread60
+#define DESCRIPTION A simple copycat
```
```suggestion
```

noclew replied:
done

ridingqwerty commented on keyboards/nckiibs/flatbread60/config.h:
```diff
@@ -0,0 +1,251 @@
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
+/*
+ * HD44780 LCD Display Configuration
+ */
+/*
+#define LCD_LINES           2     //< number of visible lines of the display
+#define LCD_DISP_LENGTH    16     //< visibles characters per line of the display
+
+#define LCD_IO_MODE      1            //< 0: memory mapped mode, 1: IO port mode
+
+#if LCD_IO_MODE
+#define LCD_PORT         PORTB        //< port for the LCD lines
+#define LCD_DATA0_PORT   LCD_PORT     //< port for 4bit data bit 0
+#define LCD_DATA1_PORT   LCD_PORT     //< port for 4bit data bit 1
+#define LCD_DATA2_PORT   LCD_PORT     //< port for 4bit data bit 2
+#define LCD_DATA3_PORT   LCD_PORT     //< port for 4bit data bit 3
+#define LCD_DATA0_PIN    4            //< pin for 4bit data bit 0
+#define LCD_DATA1_PIN    5            //< pin for 4bit data bit 1
+#define LCD_DATA2_PIN    6            //< pin for 4bit data bit 2
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
+/* Bootmagic Lite key configuration */
+// #define BOOTMAGIC_LITE_ROW 0
+// #define BOOTMAGIC_LITE_COLUMN 0
```
```suggestion
```

noclew replied:
done!

ridingqwerty commented on keyboards/nckiibs/flatbread60/config.h:
```diff
@@ -0,0 +1,251 @@
+/*
+ * Split Keyboard specific options, make sure you have 'SPLIT_KEYBOARD = yes' in your rules.mk, and define SOFT_SERIAL_PIN.
+ */
+#define SOFT_SERIAL_PIN D0  // or D1, D2, D3, E6
```
```suggestion
//#define SOFT_SERIAL_PIN D0  // or D1, D2, D3, E6
```

noclew replied:
done!

noclew commented:
Thank you very much for your help! I fixed the issues pointed out.

noclew commented:
I also added two more keyboards. Do I have to close and make another pull request?

drashna commented on keyboards/nckiibs/vaneelaex/rules.mk:
```diff
@@ -0,0 +1,23 @@
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+# FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+# HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
```suggestion
```

noclew commented:
All done!

drashna commented on keyboards/nckiibs/flatbread60/config.h:
```diff
@@ -0,0 +1,175 @@
+#define VENDOR_ID 0x9906
+#define PRODUCT_ID 0x0000
+#define DEVICE_VER 0x0001
+#define MANUFACTURER noclew
```
The folder that this is in is `nckiibs` but manu is `noclew`?

noclew replied:
Oh, I can change it to ncKiibs. Done!

drashna commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,212 @@
+  state = update_tri_layer_state(state, _RAISE, _LOWER, _ADJUST);
+  return state;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        print("mode just switched to qwerty and this is a huge string\n");
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_COLEMAK);
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+      break;
+    case BACKLIT:
+      if (record->event.pressed) {
+        register_code(KC_RSFT);
+        #ifdef BACKLIGHT_ENABLE
+          backlight_step();
+        #endif
+        PORTE &= ~(1<<6);
+      } else {
+        unregister_code(KC_RSFT);
+        PORTE |= (1<<6);
+      }
+      return false;
+      break;
+  }
+  return true;
```
This is just a nitpick, but this block could use a clang-format pass

noclew replied:
Can you let me know how I can do so? Sorry, I am sure what is a clang-format pass. :(

noclew replied:
Oh, if you are talking about formatting, I just fixed it. Thanks!

noclew replied:
Done!

drashna commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,120 @@
+
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  state = update_tri_layer_state(state, _RAISE, _LOWER, _FN);
+  return state;
+}
```
I believe that VIA uses some custom keys to facilitate this. 

And specifically, having it here means that you CANNOT disable this functionality on the board, at all.  Which may be less than ideal.

noclew replied:
I see. I deleted these lines. Done!

drashna commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,107 @@
+
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+  state = update_tri_layer_state(state, _RAISE, _LOWER, _FN);
+  return state;
+}
```
Same as above.

noclew replied:
Done!

drashna commented on keyboards/nckiibs/vaneelaex/vaneelaex.h:
```diff
@@ -0,0 +1,42 @@
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define LAYOUT_ss_6x12( \
```
This layout has a 2U spacebar in the center, but is otherwise ortho? 
If so, it may be better to use `LAYOUT_ortho_6x12_1x2uC` for the layout name. 

(but I'm horrible at naming things.

noclew replied:
Oh, it is semi-staggered, so I just put ss. If it is not appropriate, please let me know!

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ * */
+[_QWERTY] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,  \
+  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
[_QWERTY] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  KC_NO, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * `-----------------------------------------------------------------------------------'
+ */
+
+[_COLEMAK] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_F,    KC_P,    KC_G,    KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN, KC_DEL,  \
+  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,    KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,  \
+  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
[_COLEMAK] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_F,    KC_P,    KC_G,    KC_J,    KC_L,    KC_U,    KC_Y,    KC_SCLN, KC_DEL,
  KC_ESC,  KC_A,    KC_R,    KC_S,    KC_T,    KC_D,    KC_H,    KC_N,    KC_E,    KC_I,    KC_O,    KC_QUOT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_K,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
  KC_NO, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_DVORAK] = LAYOUT_ortho_5x12( \
+    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,    KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,    KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH, \
+  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,    KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT,  \
+  BACKLIT, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
[_DVORAK] = LAYOUT_ortho_5x12(
    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_QUOT, KC_COMM, KC_DOT,  KC_P,    KC_Y,    KC_F,    KC_G,    KC_C,    KC_R,    KC_L,    KC_DEL,
  KC_ESC,  KC_A,    KC_O,    KC_E,    KC_U,    KC_I,    KC_D,    KC_H,    KC_T,    KC_N,    KC_S,    KC_SLSH,
  KC_LSFT, KC_SCLN, KC_Q,    KC_J,    KC_K,    KC_X,    KC_B,    KC_M,    KC_W,    KC_V,    KC_Z,    KC_ENT,
  KC_NO, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_ortho_5x12( \
+  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  \
+  _______, RESET,   DEBUG,   _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______, \
+  _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_ADJUST] = LAYOUT_ortho_5x12(
  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,
  _______, RESET,   DEBUG,   _______, _______, _______, _______, TERM_ON, TERM_OFF,_______, _______, KC_DEL,
  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, QWERTY,  COLEMAK, DVORAK,  _______, _______,
  _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+            break;
```
```suggestion
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            return false;
+            break;
```
```suggestion
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+                set_single_persistent_default_layer(_DVORAK);
+            }
+            return false;
+            break;
```
```suggestion
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ * */
+    [_BASE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,  \
+  KC_Z , KC_LCTL, KC_LALT, KC_LGUI,   LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
    [_BASE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
  KC_Z , KC_LCTL, KC_LALT, KC_LGUI,   LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ortho_5x12( \
+  RESET  , DEBUG  , _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,  \
+  _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______, \
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_FN] = LAYOUT_ortho_5x12(
  RESET  , DEBUG  , _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,
  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/flatbread60/rules.mk:
```diff
@@ -0,0 +1,23 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = caterina
```
```suggestion
# Bootloader selection
BOOTLOADER = caterina
```

fauxpark commented on keyboards/nckiibs/vaneela/config.h:
```diff
@@ -0,0 +1,164 @@
+#define DEVICE_VER 0x0001
+#define MANUFACTURER noclew
+#define PRODUCT Vaneela
+#define DESCRIPTION A semi-staggered 60 keyboard
```
```suggestion
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * | Ctrl | Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ * */
+    [_BASE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,  \
+  KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
    [_BASE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
  KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_BSPC, KC_9,    KC_0,    KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, _______, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_BSPC, KC_9,    KC_0,    KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, _______, KC_BSLS,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ortho_5x12( \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,  \
+  _______, RESET, DEBUG, _______, _______, _______, _______, _______, _______,TERM_ON, TERM_OFF, KC_DEL,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______, \
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_FN] = LAYOUT_ortho_5x12(
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RESET, DEBUG, _______, _______, _______, _______, _______, _______,TERM_ON, TERM_OFF, KC_DEL,
  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ * */
+    [_BASE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,  \
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT, \
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,  \
+  KC_Z , KC_LCTL, KC_LALT, KC_LGUI,   LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
    [_BASE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
  KC_Z , KC_LCTL, KC_LALT, KC_LGUI,   LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ortho_5x12( \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC, \
+  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ortho_5x12(
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
  KC_TILD, KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ortho_5x12( \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC, \
+  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ortho_5x12(
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_BSPC,
  KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,115 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ortho_5x12( \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,  \
+  _______, RESET, DEBUG, _______, _______, _______, _______, _______, _______,_______, _______, _______,  \
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______, \
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_FN] = LAYOUT_ortho_5x12(
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
  _______, RESET, DEBUG, _______, _______, _______, _______, _______, _______,_______, _______, _______,
  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/vaneela/rules.mk:
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = caterina
```
```suggestion
# Bootloader selection
BOOTLOADER = caterina
```

fauxpark commented on keyboards/nckiibs/vaneelaex/config.h:
```diff
@@ -0,0 +1,163 @@
+#define DEVICE_VER 0x0001
+#define MANUFACTURER nckiibs
+#define PRODUCT VaneelaEx
+#define DESCRIPTION A Semi-staggered 40 keyboard with number keys
```
```suggestion
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * '---------------------------------------------------------------------------------------------------------------'
+ * */
+[_BASE] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT, \
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,  \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
[_BASE] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_TILDE, KC_9,    KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_DEL,   KC_6,    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+KC_KP_PLUS,     KC_1,           KC_2,           _______,  KC_3,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      _______,  KC_0,    _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_TILDE, KC_9,    KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_DEL,   KC_6,    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
KC_KP_PLUS,     KC_1,           KC_2,           _______,  KC_3,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      _______,  KC_0,    _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ss_6x12( \
+  KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ss_6x12(
  KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      |      |      |      |      |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______, \
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_FN] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |  /   |  *   |  .   |   0  | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * '---------------------------------------------------------------------------------------------------------------'
+ * */
+[_BASE] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT, \
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,  \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT  \
```
```suggestion
[_BASE] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_TILDE, KC_9,    KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_DEL,   KC_6,    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, \
+KC_KP_PLUS,     KC_1,           KC_2,           _______,  KC_3,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______, \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      _______,  KC_0,    _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY \
```
```suggestion
[_LOWER] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_TILDE, KC_9,    KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_DEL,   KC_6,    KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE,
KC_KP_PLUS,     KC_1,           KC_2,           _______,  KC_3,    KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,S(KC_NUHS),S(KC_NUBS),KC_HOME, KC_END, _______,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      _______,  KC_0,    _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ss_6x12( \
+  KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,  \
+  KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______, \
+  KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY  \
```
```suggestion
[_RAISE] = LAYOUT_ss_6x12(
  KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
  KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
  KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, KC_PGUP, KC_PGDN, _______,
  KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    _______, _______, _______, _______, _______, _______, _______, _______, KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,102 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      |      |      |      |      |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12( \
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,  \
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______, \
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______, \
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______  \
```
```suggestion
[_FN] = LAYOUT_ss_6x12(
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
```

fauxpark commented on keyboards/nckiibs/vaneelaex/rules.mk:
```diff
@@ -0,0 +1,21 @@
+# MCU name
+MCU = atmega32u4
+
+BOOTLOADER = caterina
```
```suggestion
# Bootloader selection
BOOTLOADER = caterina
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+            }
+            return false;
+            break;
+        case BACKLIT:
+            if (record->event.pressed) {
+                register_code(KC_RSFT);
+#ifdef BACKLIGHT_ENABLE
+                backlight_step();
+#endif
+                PORTE &= ~(1 << 6);
+            } else {
+                unregister_code(KC_RSFT);
+                PORTE |= (1 << 6);
+            }
+            return false;
+            break;
```
```suggestion
```

noclew replied:
done

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+    QWERTY = SAFE_RANGE,
+    COLEMAK,
+    DVORAK,
+
+    BACKLIT,
```
```suggestion
```

noclew replied:
done

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
```
```suggestion
 * |      | Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
```

noclew replied:
done

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   K  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
```
```suggestion
 * |      | Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
```

noclew replied:
done

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,205 @@
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   ;  |   Q  |   J  |   K  |   X  |   B  |   M  |   W  |   V  |   Z  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
```
```suggestion
 * |      | Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
```

noclew replied:
done

noroadsleft commented on keyboards/nckiibs/vaneela/readme.md:
```diff
@@ -0,0 +1,19 @@
+# Vaneela
+
+![vaneela](https://i.imgur.com/wp64G8v.jpg)
```
```suggestion
![vaneela](https://i.imgur.com/wp64G8vl.jpg)
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/config.h:
```diff
@@ -0,0 +1,163 @@
+/* defined by default; to change, uncomment and set to the combination you want */
+// #define IS_COMMAND() (get_mods() == MOD_MASK_SHIFT)
+
+/* control how magic key switches layers */
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_FKEYS  true
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_NKEYS  true
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_CUSTOM false
+
+/* override magic key keymap */
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_FKEYS
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_NKEYS
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_CUSTOM
+//#define MAGIC_KEY_HELP           H
+//#define MAGIC_KEY_HELP_ALT       SLASH
+//#define MAGIC_KEY_DEBUG          D
+//#define MAGIC_KEY_DEBUG_MATRIX   X
+//#define MAGIC_KEY_DEBUG_KBD      K
+//#define MAGIC_KEY_DEBUG_MOUSE    M
+//#define MAGIC_KEY_VERSION        V
+//#define MAGIC_KEY_STATUS         S
+//#define MAGIC_KEY_CONSOLE        C
+//#define MAGIC_KEY_LAYER0         0
+//#define MAGIC_KEY_LAYER0_ALT     GRAVE
+//#define MAGIC_KEY_LAYER1         1
+//#define MAGIC_KEY_LAYER2         2
+//#define MAGIC_KEY_LAYER3         3
+//#define MAGIC_KEY_LAYER4         4
+//#define MAGIC_KEY_LAYER5         5
+//#define MAGIC_KEY_LAYER6         6
+//#define MAGIC_KEY_LAYER7         7
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
```suggestion
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/info.json:
```diff
@@ -0,0 +1,326 @@
+    "width": 16.5,
+    "height": 4,
+    "layouts": {
+        "LAYOUT": {
```
```suggestion
        "LAYOUT_ss_6x12": {
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/readme.md:
```diff
@@ -0,0 +1,20 @@
+# Vaneela Ex
+
+![vaneelEx](https://i.imgur.com/5QmJLbj.jpg)
```
```suggestion
![vaneelEx](https://i.imgur.com/5QmJLbjl.jpg)
```

noclew commented on keyboards/nckiibs/flatbread60/keymaps/via/rules.mk:
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+RGBLIGHT_ENABLE = no
```
oh, this one does not have extra rgb pins. I will fix it.

noclew commented:
> In addition to the specific suggestions, I'm not keen on the fact that the `default` and `via` keymaps for each board don't use the same mappings. This can be disorienting to users who switch between VIA and non-VIA firmware, and IMO these keymaps should be as similar as possible. I may return with more suggestions along these lines later.
> 
> GitHub tip: Multiple suggestions can be applied to a single commit through the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/9817/files).

Thank you! I fixed what were requested. By the way, what is the proper response to the change requests? Do I have to click every 'resolve conversation' button after fixing them?

noclew commented on keyboards/nckiibs/flatbread60/config.h:
```diff
@@ -0,0 +1,175 @@
+/* defined by default; to change, uncomment and set to the combination you want */
+// #define IS_COMMAND() (get_mods() == MOD_MASK_SHIFT)
+
+/* control how magic key switches layers */
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_FKEYS  true
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_NKEYS  true
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_CUSTOM false
+
+/* override magic key keymap */
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_FKEYS
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_NKEYS
+//#define MAGIC_KEY_SWITCH_LAYER_WITH_CUSTOM
+//#define MAGIC_KEY_HELP           H
+//#define MAGIC_KEY_HELP_ALT       SLASH
+//#define MAGIC_KEY_DEBUG          D
+//#define MAGIC_KEY_DEBUG_MATRIX   X
+//#define MAGIC_KEY_DEBUG_KBD      K
+//#define MAGIC_KEY_DEBUG_MOUSE    M
+//#define MAGIC_KEY_VERSION        V
+//#define MAGIC_KEY_STATUS         S
+//#define MAGIC_KEY_CONSOLE        C
+//#define MAGIC_KEY_LAYER0         0
+//#define MAGIC_KEY_LAYER0_ALT     GRAVE
+//#define MAGIC_KEY_LAYER1         1
+//#define MAGIC_KEY_LAYER2         2
+//#define MAGIC_KEY_LAYER3         3
+//#define MAGIC_KEY_LAYER4         4
+//#define MAGIC_KEY_LAYER5         5
+//#define MAGIC_KEY_LAYER6         6
+//#define MAGIC_KEY_LAYER7         7
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
changed!

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,190 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_ortho_5x12( \
```
```suggestion
[_ADJUST] = LAYOUT_ortho_5x12(
```

fauxpark commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,159 @@
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_ortho_5x12( \
```
```suggestion
[_ADJUST] = LAYOUT_ortho_5x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * '---------------------------------------------------------------------------------------------------------------'
+ * */
+[_BASE] = LAYOUT_ss_6x12( \
```
```suggestion
[_BASE] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ss_6x12( \
```
```suggestion
[_LOWER] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ss_6x12( \
```
```suggestion
[_RAISE] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      |      |      |      |      |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12( \
```
```suggestion
[_FN] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+ * |  /   |  *   |  .   |   0  | Brite| Ctrl | Alt  | GUI  |Lower | Space|Space |Raise | Left | Down |  Up  |Right |
+ * '---------------------------------------------------------------------------------------------------------------'
+ * */
+[_BASE] = LAYOUT_ss_6x12( \
```
```suggestion
[_BASE] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_ss_6x12( \
```
```suggestion
[_LOWER] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_ss_6x12( \
```
```suggestion
[_RAISE] = LAYOUT_ss_6x12(
```

fauxpark commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+ * |  /   |  *   |  .   |   0  |      |      |      |      |      |             |      |      |      |      |      |
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12( \
```
```suggestion
[_FN] = LAYOUT_ss_6x12(
```

noroadsleft commented:
> Thank you! I fixed what were requested. By the way, what is the proper response to the change requests? Do I have to click every 'resolve conversation' button after fixing them?

Depends how you apply the suggestions. If you apply them through GitHub's web interface, they should resolve automatically. If you make the changes locally and then push them to your fork, you do need to resolve them manually.

With a lot of suggestions, I'd suggest using the Files Changed tab to make them, then when you have them committed on GitHub, use `git pull origin nckiibs` to pull those changes to the source branch on your local repo (the files on your computer). That will automatically resolve the suggestions, but keep your fork and your local repository synchronized. Note that you'll need to have your local `nckiibs` branch checked out and clean (no modifications) for this to work without any hiccups/issues.

noroadsleft commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+[_FN] = LAYOUT_ortho_5x12(
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+  _______, RESET,   DEBUG,   _______, _______, _______, _______, _______, _______,TERM_ON, TERM_OFF, KC_DEL,
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
```
A bunch of keycodes here require the Audio and Midi features, neither of which are enabled.

```suggestion
  _______, _______, _______, _______, _______, AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

noroadsleft commented on keyboards/nckiibs/vaneela/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,115 @@
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
```
Enables access to the `_FN` layer.

```suggestion
};

layer_state_t layer_state_set_user(layer_state_t state) {
  state = update_tri_layer_state(state, _RAISE, _LOWER, _FN);
  return state;
}
```

noroadsleft commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,114 @@
+[_FN] = LAYOUT_ortho_5x12(
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+  _______, RESET,   DEBUG,   _______, _______, _______, _______, _______, _______,TERM_ON, TERM_OFF, KC_DEL,
+  _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
+  _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
```
```suggestion
  _______, _______, _______, _______, _______, AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

noroadsleft commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,114 @@
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
```
Enables access to the `_FN` layer.

```suggestion
};

layer_state_t layer_state_set_user(layer_state_t state) {
  state = update_tri_layer_state(state, _RAISE, _LOWER, _FN);
  return state;
}
```

noclew replied:
@drashna told me that VIA has some custom keycodes to switch layers, so I was asked to remove these lines in the last request. Can you clarify what I should follow, please?
https://github.com/qmk/qmk_firmware/pull/9817#discussion_r464779728

noroadsleft commented on keyboards/nckiibs/vaneelaex/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,108 @@
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12(
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
```
```suggestion
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______, _______, _______, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, _______, _______, _______, AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+ * `---------------------------------------------------------------------------------------------------------------'
+ */
+[_FN] = LAYOUT_ss_6x12(
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______,TERM_ON, TERM_OFF, KC_DEL,
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, MU_MOD,  AU_ON,   AU_OFF,  AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
```
```suggestion
KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,   RESET  , RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAI, RGB_VAD, RGB_SAI, RGB_SAD, _______, _______, _______, KC_DEL,
KC_KP_MINUS,    KC_4,           KC_5,           KC_6,   _______, _______, _______, _______, _______, AG_NORM, AG_SWAP, _______, _______, _______, _______, _______,
KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,   _______, MUV_DE,  MUV_IN,  MU_ON,   MU_OFF,  MI_ON,   MI_OFF,  _______, _______, _______, _______, _______,
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+};
```
```suggestion
};

layer_state_t layer_state_set_user(layer_state_t state) {
  state = update_tri_layer_state(state, _RAISE, _LOWER, _FN);
  return state;
}
```

noclew replied:
@drashna told me that VIA has some custom keycodes to switch layers, so I was asked to remove these lines in the last request. Can you clarify what I should follow, please?
https://github.com/qmk/qmk_firmware/pull/9817#discussion_r464779728

noclew commented:
> > Thank you! I fixed what were requested. By the way, what is the proper response to the change requests? Do I have to click every 'resolve conversation' button after fixing them?
> 
> Depends how you apply the suggestions. If you apply them through GitHub's web interface, they should resolve automatically. If you make the changes locally and then push them to your fork, you do need to resolve them manually.
> 
> With a lot of suggestions, I'd suggest using the Files Changed tab to make them, then when you have them committed on GitHub, use `git pull origin nckiibs` to pull those changes to the source branch on your local repo (the files on your computer). That will automatically resolve the suggestions, but keep your fork and your local repository synchronized. Note that you'll need to have your local `nckiibs` branch checked out and clean (no modifications) for this to work without any hiccups/issues.

@noroadsleft @drashna 
Thank you for the information. I am learning this review process, so it will be appreciated if you can answer these noob questions!

1. Upon the change requests by reviewers, is there a way to filter the files that were quested to change in the `Files Changed` tab? Last time, I went there and added, ran batch commits, and marked all files views. After the new change requests were delivered, I check the Files Changed tab, and because all the files were still marked as 'viewed', there was no way to filter out the files that the changes have been requested with. In this process, is there a way to quickly check the files with change requests? Or, do I have to go through every file again and check the red box to find the new change requests?

2. Once I changed the files requested and batch committed them on Github, I more than often have failures in the automated checks because of the files of other keyboards. This time, it is PR Lint keyboards check, and it looks like there are some errors with the files of other keyboards. If this happened, what I did was 1) pull the lastest Github batch commit to my local repo 2) rebase the local repo with the latest QMK build 3) commit and push the local repo to Github and wait to see if it passes the two automated checks. Is this procedure correct to deal with the automated check failures? Sometimes this did not work when the official QMK repo is not updated and is still with errors, so I feel like this is not the correct way to deal with the failures in automated checks from others' files. Can you let me know what is the proper procedure?

3. I am a bit confused by VIA keymaps. @drashna told me that functions for switching layers are not necessary because VIA has its own. I looked at others' file, and it seems like they do not define custom functions to switch layers and only have the keymaps. However, in the latest change request, it was asked to put the functions back to the keymap files, so I wonder what is the right way to write the VIA kemaps. 

I am sorry for the long questions, and I really appreciate all the efforts to help newbies in the community!

drashna commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,159 @@
+  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+)
+
+};
```
VIA actually some custom keycodes for handling tri layer stuff.

noclew replied:
> VIA actually some custom keycodes for handling tri layer stuff.

Can you let me know where I can find the information about it? I could not find it in the official doc.

noroadsleft commented:
> 1. Upon the change requests by reviewers, is there a way to filter the files that were quested to change in the `Files Changed` tab? Last time, I went there, added changes to the batch, ran batch commits, and marked all files as `viewed`. After the new changes were requested, I checked the `Files Changed tab` again, and because all the files were still marked as 'viewed', there was no way to filter out the files that the new changes have been requested with. In this process, is there a way to quickly filter out the files with new change requests? Or, do I have to go through every file again and check the red box to find the new change requests?

I don't have an answer for this one; I'll probably have to look into it. I know what things are likely to be requested due to being a collaborator, so I typically jump ahead on these things before opening a PR.

> 2. Once I changed the files requested and batch committed them on Github, I more than often have failures in the automated checks because of the files of other keyboards. This time, it is PR Lint keyboards check, and it looks like there are some errors with the files of other keyboards, specifically, `Run qmk info` test with `Kudox` keyboard. If this happens, what I usually do is 1) pull the lastest Github batch commit to my local repo 2) rebase the local repo with the latest QMK build 3) commit and push the local repo to Github and 4) wait to see if it passes the two automated checks. Is this procedure correct to deal with the automated check failures? Sometimes this does not work when the official QMK repo is not updated and is still with errors, so I feel like this is not the correct way to deal with the failures in automated checks from others' files. Can you let me know what is the proper procedure?

The PR Lint check is literally a "first-effort" – if it's not flagging boards you've changed, ignore it.

> 3. I am a bit confused by VIA keymaps. @drashna told me that functions for switching layers are not necessary because VIA has its own. I looked at others' file, and it seems like they do not define custom functions to switch layers and only have the keymaps. However, in the latest change request, it was asked to put the functions back to the keymap files, so I wonder what is the right way to write the VIA kemaps.

VIA's functions for this only work if you use [the keycodes VIA provides for doing this](https://github.com/qmk/qmk_firmware/blob/2d8703f6cc61918aaefd3ce04a064a82ab238140/quantum/via.h#L108-L109), and they also only work if you have exactly four layers.

noclew commented:
> > 1. Upon the change requests by reviewers, is there a way to filter the files that were quested to change in the `Files Changed` tab? Last time, I went there, added changes to the batch, ran batch commits, and marked all files as `viewed`. After the new changes were requested, I checked the `Files Changed tab` again, and because all the files were still marked as 'viewed', there was no way to filter out the files that the new changes have been requested with. In this process, is there a way to quickly filter out the files with new change requests? Or, do I have to go through every file again and check the red box to find the new change requests?
> 
> I don't have an answer for this one; I'll probably have to look into it. I know what things are likely to be requested due to being a collaborator, so I typically jump ahead on these things before opening a PR.
> 
> > 1. Once I changed the files requested and batch committed them on Github, I more than often have failures in the automated checks because of the files of other keyboards. This time, it is PR Lint keyboards check, and it looks like there are some errors with the files of other keyboards, specifically, `Run qmk info` test with `Kudox` keyboard. If this happens, what I usually do is 1) pull the lastest Github batch commit to my local repo 2) rebase the local repo with the latest QMK build 3) commit and push the local repo to Github and 4) wait to see if it passes the two automated checks. Is this procedure correct to deal with the automated check failures? Sometimes this does not work when the official QMK repo is not updated and is still with errors, so I feel like this is not the correct way to deal with the failures in automated checks from others' files. Can you let me know what is the proper procedure?
> 
> The PR Lint check is literally a "first-effort" – if it's not flagging boards you've changed, ignore it.
> 
> > 1. I am a bit confused by VIA keymaps. @drashna told me that functions for switching layers are not necessary because VIA has its own. I looked at others' file, and it seems like they do not define custom functions to switch layers and only have the keymaps. However, in the latest change request, it was asked to put the functions back to the keymap files, so I wonder what is the right way to write the VIA kemaps.
> 
> VIA's functions for this only work if you use [the keycodes VIA provides for doing this](https://github.com/qmk/qmk_firmware/blob/2d8703f6cc61918aaefd3ce04a064a82ab238140/quantum/via.h#L108-L109), and they also only work if you have exactly four layers.

Thanks for the reply. I checked other keyboards like Contra, and they were only assigned MO(<LayerName>) keys in VIA. I followed their ways and changed the number of layers to four as you suggested.

noroadsleft commented on keyboards/nckiibs/flatbread60/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,114 @@
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_DEL,
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_ENT,
+  KC_NO, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```
Using the VIA keycodes to access the adjust layer. Without this change there is no way to access the `_ADJUST` layer.

```suggestion
  KC_NO, KC_LCTL, KC_LALT, KC_LGUI, FN_MO13, KC_SPC,  KC_SPC,  FN_MO23, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

noroadsleft commented on keyboards/nckiibs/vaneela/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,114 @@
+  KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+  KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
+  KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
+  KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,   KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```
As above.

```suggestion
  KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, FN_MO13, KC_SPC,  KC_SPC,  FN_MO23, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

noroadsleft commented on keyboards/nckiibs/vaneelaex/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,103 @@
+KC_KP_EQUAL,    KC_7,           KC_8,           KC_9,    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+KC_KP_MINUS,    KC_4,           KC_5,           KC_6,    KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_ENT,
+KC_KP_PLUS,     KC_1,           KC_2,           KC_3,    KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_QUOT,
+KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, LOWER,   KC_SPC,  KC_SPC,  RAISE,    KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```
As above.

```suggestion
KC_KP_SLASH,    KC_KP_ASTERISK, KC_KP_DOT,      KC_0,    KC_LCTL, KC_LCTL, KC_LALT, KC_LGUI, FN_MO13, KC_SPC,  KC_SPC,  FN_MO23, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT
```

noroadsleft commented:
Thanks!

noclew commented:
Thank you!


On Sun, Aug 9, 2020 at 10:51 AM James Young <notifications@github.com>
wrote:

> Thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qmk/qmk_firmware/pull/9817#issuecomment-671061229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOXY4HHIFWLUMZCVDJ3WBDR72ZVTANCNFSM4PHG2QUQ>
> .
>

noclew commented:
Oh, by the way, I just checked qmk configurator, and it seems like the
default keymap was no registered. How can I make it so that the there shows
a default keymap in QMK configurator?

On Sun, Aug 9, 2020 at 10:51 AM James Young <notifications@github.com>
wrote:

> Thanks!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qmk/qmk_firmware/pull/9817#issuecomment-671061229>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOXY4HHIFWLUMZCVDJ3WBDR72ZVTANCNFSM4PHG2QUQ>
> .
>

noroadsleft commented:
> Oh, by the way, I just checked qmk configurator, and it seems like the default keymap was no registered. How can I make it so that the there shows a default keymap in QMK configurator?

@noclew,

QMK Configurator keymaps live here: https://github.com/qmk/qmk_configurator/tree/master/public/keymaps

Instructions: https://github.com/qmk/qmk_firmware/blob/633d0fe5f7eef8400cfdc843f1a043d0eac8a836/docs/configurator_default_keymaps.md (hopefully this will hit the QMK docs soon)
